tests/api/test_flows.py

```diff
@@ -514,6 +516,22 @@ async def test_create_flow_persists_serialized_flow(self, project_id, flow):
         # confirm the keys in the serialized flow match the form we'd expect
         assert persisted_flow.serialized_flow == flow.serialize()
 
+    @pytest.mark.parametrize("model_name", ["Task", "Edge"])
+    async def test_create_flow_cleans_up_if_task_or_edge_creation_fails(
+        self, project_id, flow, monkeypatch, model_name
+    ):
+        patched_insert_raises_error = MagicMock(side_effect=Exception())
+        monkeypatch.setattr(
+            f"prefect.models.{model_name}.insert_many", patched_insert_raises_error
+        )
+        flow.name = "my special flow"
```

Comment:
jw - did you change the name on the flow to ensure the name was unique?

Reply:
yes, I wasn't sure what precisely our pytest logic for setup/teardown during tests was